x/sync/firewood/sync_db.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package firewood
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/ava-labs/firewood-go-ethhash/ffi"
+	"github.com/ava-labs/libevm/core/types"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/network/p2p"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/maybe"
+
+	xsync "github.com/ava-labs/avalanchego/x/sync"
+)
+
+var _ xsync.DB[*RangeProof, *ChangeProof] = (*db)(nil)
+
+// db wraps a Firewood FFI database to implement the xsync.DB interface.
+type db struct {
+	db   *ffi.Database
+	lock sync.Mutex // TODO: remove this lock once FFI is thread-safe
+}
+
+type Config struct {
+	RangeProofClient      *p2p.Client
+	ChangeProofClient     *p2p.Client
+	SimultaneousWorkLimit int
+	Log                   logging.Logger
+	TargetRoot            ids.ID
+	StateSyncNodes        []ids.NodeID
+}
+
+func NewSyncer(fw *ffi.Database, config Config, register prometheus.Registerer) (*xsync.Syncer[*RangeProof, *ChangeProof], error) {
```

Comment:
1. nit: `fw` is deviating from the abstractions of `ffi` and `Database` - would calling this something like `db` make more sense? Similarly `register` might want to be `registerer` or `r`.
2. Locality might want to be after all of the function definitions on `db` - currently this locality reads like a constructor for `db`